web_ui/src/pages/create-project/components/pose-template/sample-image/load-file-button.component.tsx

```diff
@@ -4,23 +4,36 @@
 import { Button, FileTrigger, Flex } from '@geti/ui';
 import { Image } from '@geti/ui/icons';
 
-import { onValidFileList } from '../../../../../shared/utils';
+import { NOTIFICATION_TYPE } from '../../../../../notification/notification-toast/notification-type.enum';
+import { useNotification } from '../../../../../notification/notification.component';
+import { mediaExtensionHandler } from '../../../../../providers/media-upload-provider/media-upload.validator';
+import { loadImageFromFile, VALID_IMAGE_TYPES_SINGLE_UPLOAD } from '../../../../../shared/media-utils';
+import { onValidImageFormat } from '../../../../utils';
 
 interface LoadFileButtonProps {
     onFileLoaded: (image: string) => void;
 }
 
+export const errorMessage =
+    'Only image files are supported for this feature. Allowed formats: ' +
+    mediaExtensionHandler(VALID_IMAGE_TYPES_SINGLE_UPLOAD);
+
 export const LoadFileButton = ({ onFileLoaded }: LoadFileButtonProps) => {
-    const onProcessUploadFile = onValidFileList(async ([file]: File[]) => {
-        const reader = new FileReader();
-        reader.onloadend = () => {
-            onFileLoaded(String(reader.result));
-        };
-        reader.readAsDataURL(file);
-    });
+    const { addNotification } = useNotification();
+
+    const handleValidFormat = async ([file]: File[]) => {
+        const image = await loadImageFromFile(file);
+        onFileLoaded(image.src);
+    };
+
+    const handleInvalidFormat = () => {
+        addNotification({ message: errorMessage, type: NOTIFICATION_TYPE.ERROR });
+    };
+
+    const handleUploadFile = onValidImageFormat(handleValidFormat, handleInvalidFormat);
 
     return (
-        <FileTrigger onSelect={onProcessUploadFile} aria-label='upload sample image'>
+        <FileTrigger onSelect={handleUploadFile} aria-label='upload sample image' data-testid='upload-sample-image'>
```

Comment:
`FileTrigger` supports [`acceptedFileTypes`](https://react-spectrum.adobe.com/react-spectrum/FileTrigger.html#accepted-file-types) can we use that to prevent the user from uploading non-images?

Reply:
Theoretically, yes, but opening the dev tools and removing it from the HTML would be enough to bypass that restriction. I’ll add it, but we should keep the current validations as well.